File: Ackermann.java
public class Ackermann {
	static int m = 0;
	static int n = 0;
	public static void main(String[] args){
		
		int i = 0;
		while(true){
			int m = i;
			while(m > -1){
				System.out.println(ackermann(m, n));
				m++;
				n--;
			}
			i++;
			n = 0;
		}
	}
	
	public static int ackermann(int m, int n){
		if(m == 0){
			return n + 1;
		}
		else if(m > 0 && n == 0){
			return ackermann(m - 1, 1);
		}
		else if (m > 0 && n > 0){
			return ackermann(m - 1, ackermann(m, n - 1));
		}
		return -1;
	}
}

File: Ackermann.swift
func ackermann(_ m: Int, _ n: Int) -> Int {
    var result: Int

    if m == 0 {
        result = n + 1
    }
    else if n == 0 {
        result = ackermann(m-1, 1)
    }
    else {
        result = ackermann(m-1, ackermann(m, n-1))
    }

    return result
}

var i = 0
var n = 0

while true {
    var m = i
    while m > -1 {
        print("ackermann(\(m), \(n)) = \(ackermann(m, n))")
        m -= 1
        n += 1
    }
    i += 1
    n = 0
}
